CMakeLists.txt

```diff
@@ -23,7 +23,8 @@ set(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG")
 file(GLOB SRCS *.c)
 
 set(COMMAND_PREFIX CMD:)
-configure_file(config.h.in config.h)
+configure_file(config.h.in ${PROJECT_BINARY_DIR}/config.h)
+include_directories(BEFORE ${PROJECT_BINARY_DIR})
```

Comment:
This looks right to me, but please add a body to your commit message briefly describing how `config.h` would be created in the _source_ directory, according to the [CMake default](https://cmake.org/cmake/help/latest/command/include_directories.html) for relative paths, when building out-of-tree.

Reply:
Will do.

Reply:
Regarding the commit message, the line lengths were a little inconsistent, so I mocked up another version that also adds some extra whitespace between paragraphs.

<img width="527" height="314" alt="image" src="https://github.com/user-attachments/assets/075f6a4f-fc63-4d9c-81b6-a998fd206a69" />

Reply:
Done